test-clients/src/main/resource/contract/contracts/ExchangeRatePrecompile/RateAware.sol

```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity >=0.5.0 <0.9.0;
+
+abstract contract RateAware {
+    uint64 constant TINY_PARTS_PER_WHOLE = uint64(100_000_000);
+    bytes1 constant TO_TINYBARS = bytes1(0xbb);
+    bytes1 constant TO_TINYCENTS = bytes1(0xcc);
+    address constant precompileAddress = address(0x21D4);
+
+    function toTinybars(int64 tinycents) 
+        internal 
+        returns (bool success, int64 tinybars) 
```

Comment:
Why do we return a success and a return value?  I cannot think of a scenario where we would return false from the precompile and not want to fail the function call.  Overflow?  Should fail with revert. Exchange rate not available?  Should be failing much earlier, like before the EVM starts.  It also makes the function awkward to use.  Solidity is not Go.

Reply:
Switched [helper methods](https://github.com/hashgraph/hedera-services/blob/03301-exchange-rate-system-contract/test-clients/src/main/resource/contract/contracts/ExchangeRatePrecompile/SelfFunding.sol#L10) to revert on a failed precompile call and only return a `uint256`. (Doesn't seem to be any point to using a narrower integral type for the methods, and makes use with `msg.sender` annoying.)